### ISSUE TITLE
convert .npy segmentation file into .nrrd

### ISSUE THREAD

**Issue by silkezimm** [zhenweishi/Py-rex#12]
Dear all, 

i have the problem that i have DICOM images and a npy file of segmented data as input to pyradiomics. #
i converted the DICOM images in nrrd format with 3Dslicer. now i need to convert the segmentation/label file .npy into .nrrd. 

i cant solve this;> can anyone help me and provide a solution?

thank you so much, 
Silke

**Comment by ivanzhovannik** [comment]
Hi @silkezimm,

In order to give you good advice, could you please share more info about our `.npy` file? Given file name `filename.npy` (_change to your file name_), please run and post below:

```
import numpy as np

mask_array  = np.load("filename.npy", allow_pickle=True)
print(f"Shape of my mask array is {mask_array.shape}")
print(f"Unique mask values are {np.unique(mask_array)}")
```

Usually `.npy` files contain only the voxel intensities and not the grid, which can be used to understand how to place these voxel intensities in the space.

On the other hand, image files such as `.nrrd` contain both the grid and voxel intensities. Given your image file `image.nrrd` (_change_) please run and post below:

```
import SimpleITK as sitk

image = sitk.ReadImage("image.nrrd")
print(f"Image size {sitk.GetSize(image)}, image spacing and origin: {sitk.GetSpacing(image)}, {sitk.GetOrigin(image)}")
```

**Comment by silkezimm** [comment]
HI!

i solved this issue, thank you soo much for your kind reply :)
actually i have another issue with using pyradiomics.. i was wondering
wheter you can help me..
i am using pyradiomics from the command line.
i would like to know whether i extracted the features without param file or
any specification from the original image or the segmented region (with the
mask file..).


i am performing batch feature extraction with the command : pyradiomics
<path/to/input>
, this works nicely,

now i have created a parameter file in .yaml format, and it says on the
readthedocs homepage to use the --param argument in the command line. here
i struggle as to how exactly tell the software to use my specific file. can
you help me here and provide the exact command i shall write?

i hope you can help..

best
silke

Am So., 30. Juli 2023 um 14:03 Uhr schrieb Ivan Ye. Zhovannik <
***@***.***>:

**Comment by ivanzhovannik** [comment]
@silkezimm like any other command in cmd/bash/other shell, I guess. See the usage here: https://pyradiomics.readthedocs.io/en/latest/usage.html
See more in Python’s argparse package how to pass flags such as —params

**Comment by silkezimm** [comment]
Yes....I have read the documentation and still can't understand the --param
....
Sorry....
